nixos/modules/services/backup/borgbackup.nix

```diff
@@ -70,16 +59,25 @@ let
 
   mkBackupService = name: cfg:
     let
-      userHome = config.users.users.${cfg.user}.home;
+      userHome = if cfg.user != null then config.users.users.${cfg.user}.home else "/var/lib/borgbackup";
     in nameValuePair "borgbackup-job-${name}" {
       description = "BorgBackup job ${name}";
       path = with pkgs; [
         borgbackup openssh
       ];
       script = mkBackupScript cfg;
-      serviceConfig = {
+      serviceConfig = optionalAttrs (!isNull cfg.user) {
         User = cfg.user;
```

Comment:
This should be `"borg"` if using a dynamic user so systemd assigns a nice name to that user.

Reply:
Actually, that might conflict with the borg user for repos, so something like `"borg-dynamic"` might be better?

Reply:
That is the section for when a user is given. I've added the `borgbackup` user to the DynamicUser section below.